### PR TITLE
Feature/11 add workflow to update sdk

### PR DESCRIPTION
**Pull request by ossaleon**

### CODE REVIEW

File: .github/workflows/update_client.yaml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      # Step 1: Checkout the first repository
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          repository: wordlift/docs
+          path: "./docs"
+
+      # Step 2: Checkout the second repository
+      - name: Checkout second repository
+        uses: actions/checkout@v3
```
Comment (ziodave) [inline .github/workflows/update_client.yaml:31]: Can we update versions here, if you open https://github.com/wordlift/docs/actions/runs/9252847204, you'll see warnings:

image

File: .github/workflows/update_client.yaml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    runs-on: ubuntu-latest
+
+    # Define the steps in the job
+    steps:
+      # Step 1: Checkout the first repository
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          repository: wordlift/docs
+          path: "./docs"
```
Comment (ziodave) [inline .github/workflows/update_client.yaml:27]: I don't think you need this, being the current repo, e.g. https://github.com/wordlift/docs/blob/358834a2db5fd4aa7ff6b07f802b2b7903d75cff/.github/workflows/azure-static-web-apps-happy-pebble-0cf31410f.yml#L14

File: .github/workflows/update_client.yaml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      - main
+    paths:
+      - "api/**"
+  pull_request:
+    types: [closed]
+    paths:
+      - "api/**"
+
+jobs:
+  run-commands:
```
Comment (ziodave) [inline .github/workflows/update_client.yaml:16]: Maybe use here a descriptive name, e.g. build_sdk

File: .github/workflows/update_client.yaml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "api/**"
+  pull_request:
+    types: [closed]
+    paths:
+      - "api/**"
```
Comment (ziodave) [inline .github/workflows/update_client.yaml:13]: Why do we need to build the SDK when a PR is closed? A closed PR is either merged to _main_ (which falls in the previous case) or to _develop_ which I think we're not interested in using to build an SDK?

File: .github/workflows/update_client.yaml [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        uses: actions/checkout@v3
+        with:
+          repository: wordlift/docs
+          path: "./docs"
+
+      # Step 2: Checkout the second repository
+      - name: Checkout second repository
+        uses: actions/checkout@v3
+        with:
+          repository: wordlift/wordlift-sdk
```
Comment (ziodave) [inline .github/workflows/update_client.yaml:33]: This is not a good repo name. Maybe python-sdk.

**Review by ziodave — CHANGES_REQUESTED** [review]
I don't think this is the right approach.

I think it is correct to create a public repo for the python sdk which must have a proper name, eg python-client (since we decided to call it wordlift-client?).

I wouldn't call the repo wordlift-sdk or wordlift-client because it is already in our wordlift org and what if we want to create a Java client/sdk.

The repo will host the source codes for the client. The repo should be responsible for generating the code and push the client to pypi.

So that the docs repo responsibility should only to push yaml updates to the python-client repo.

And the python-client repo should be responsible for merging and post processing them and then use them to generate the client, run UTs on it and push it to pypi.

Let me know if it makes sense.

File: .github/workflows/update_client.yaml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+      # Step 2: Checkout the second repository
+      - name: Checkout second repository
+        uses: actions/checkout@v4
+        with:
+          repository: wordlift/python-client
+          path: "./python-client"
+
+      # step 3: Patch middleware api specification
+      - name: Apply patch to middleware.yaml
```
Comment (ziodave) [inline .github/workflows/update_client.yaml:29]: I am undecided whether this patch should go when the docs repo receives yaml updates independently or go to the python client repo.

I am leaning towards the first, that is the docs repo should patch the yaml independently.

What do you think?

Comment (ossaleon) [reply .github/workflows/update_client.yaml:29]: both are acceptable, but i prefer that docs sends a correct version of the apis and in general to use the docs as reference.

**Review by ziodave — APPROVED** [review]

**Pull request merged**